File: 66-plus-one/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        // loop through the digits from the back and +1
        // if = 10, then continue, else return;

        for (int i = digits.size() - 1; i >= 0; i--) {
            int curr = digits[i];
            int pres = curr + 1;
            if (pres != 10) {
                digits[i] = pres;
                return digits;
            } else {
                digits[i] = 0;
            }
        }
        if (digits[0] == 0) {
            digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};

File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        add_one = digits[-1] + 1
        carry = add_one // 10
        digits[-1] = add_one % 10

        for i in range(len(digits) - 2, -1, -1):
            if not carry:
                break
            add_one = digits[i] + carry
            carry = add_one // 10
            digits[i] = add_one % 10
        
        if carry:
            digits.insert(0, 1)
        
        return digits